source/Nuke.Common/NukeBuild.Statics.cs

```diff
@@ -123,7 +124,10 @@ private static PathConstruction.AbsolutePath GetRootDirectory()
         private static PathConstruction.AbsolutePath GetBuildAssemblyDirectory()
         {
             var entryAssembly = Assembly.GetEntryAssembly();
-            if (entryAssembly == null || entryAssembly.GetTypes().All(x => !x.IsSubclassOf(typeof(NukeBuild))))
+
+            if (entryAssembly == null 
+                || entryAssembly.GetTypes().All(x => !x.IsSubclassOf(typeof(NukeBuild))) 
+                || !EnvironmentInfo.ValidDirectoryPath(entryAssembly.Location)) 
```

Comment:
The self-contained app doesn't have a build project so we will assign null because should be standalone in invoke.